src/Model/Resolver/ProductAlertSubscribe.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * ScandiPWA_ProductAlertsGraphQl
+ *
+ * @category    ScandiPWA
+ * @package     ScandiPWA_ProductAlertsGraphQl
+ * @author      Aleksandrs Kondratjevs <info@scandiweb.com>
+ * @copyright   Copyright (c) 2021 Scandiweb, Ltd (https://scandiweb.com)
+ */
+
+namespace ScandiPWA\ProductAlertsGraphQl\Model\Resolver;
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\ProductAlert\Model\Price as PriceAlert;
+use Magento\ProductAlert\Model\Stock as StockAlert;
+use Magento\Store\Model\StoreManagerInterface;
+
+class ProductAlertSubscribe implements ResolverInterface
+{
+    const PRODUCT_ALERT_PRICE_DROP = 'PRODUCT_ALERT_PRICE_DROP';
+
+    /**
+     * @var StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var ProductRepositoryInterface
+     */
+    protected $productRepository;
+
+    /**
+     * @var PriceAlert
+     */
+    protected $priceAlert;
+
+    /**
+     * @var StockAlert
+     */
+    protected $stockAlert;
+
+
+    /**
+     * ProductAlertSubscribe constructor
+     *
+     * @param PriceAlert $priceAlert
+     * @param ProductRepositoryInterface $productRepository
+     * @param StockAlert $stockAlert
+     * @param StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        PriceAlert $priceAlert,
+        ProductRepositoryInterface $productRepository,
+        StockAlert $stockAlert,
+        StoreManagerInterface $storeManager
+    ) {
+        $this->priceAlert = $priceAlert;
+        $this->productRepository = $productRepository;
+        $this->stockAlert = $stockAlert;
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $productId = (int)$args['productId'];
+        $customerId = (int)$context->getUserId();
```

Comment:
I see that on the frontend you don't check whether the user is authorized.
But i guess, this feature is for authorized customers only?

Reply:
This is intended behavior, like in Luma.
The frontend displays the subscribe buttons for any user. But when the unauthorized user clicks the subscribe button, it gives him a notification.